diff --git a/src/misc.cc b/src/misc.cc
index 6c70e40..026935a 100755
--- a/src/misc.cc
+++ b/src/misc.cc
@@ -8743,7 +8743,7 @@ static define_unary_function_eval (__os_version,&_os_version,_os_version_s);
 #ifdef HAVE_LIBCURL
 #include <curl/curl.h>
 #include <curl/easy.h>
-#include <curl/curlbuild.h>
+#include <curl/system.h>
   size_t write_data(void *ptr, size_t size, size_t nmemb, void *stream) {
     string data((const char*) ptr, (size_t) size * nmemb);
     *((stringstream*) stream) << data << endl;
